python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -166,3 +200,125 @@ def _(py_val):
     cdef unique_ptr[scalar] c_obj = make_string_scalar(py_val.encode())
     cdef Scalar slr = _new_scalar(move(c_obj), dtype)
     return slr
+
+
+@singledispatch
+def _from_numpy(np_val):
+    if np_error is not None:
+        raise np_error
+    raise TypeError(f"{type(np_val).__name__} cannot be converted to pylibcudf.Scalar")
+
+
+@_from_numpy.register(np.datetime64)
+@_from_numpy.register(np.timedelta64)
+def _(np_val):
+    raise NotImplementedError(
+        f"{type(np_val).__name__} is currently not supported."
+    )
+
+
+@_from_numpy.register(np.bool_)
+def _(np_val):
+    cdef DataType dtype = DataType(type_id.BOOL8)
+    cdef unique_ptr[scalar] c_obj = make_numeric_scalar(dtype.c_obj)
+    (<numeric_scalar[cbool]*>c_obj.get()).set_value(np_val)
```

Comment:
I'm kind of surprised that this is working. `np_val` is a Python object here. I guess it is transparently castable to the type required by `set_value` (the underlying C integral type for each of these functions)? I would have expected it to require you to do something like
```
cdef cbool val = np_val
(...).set_value(np_val)
```

Reply:
I added an explicit `cdef bool` declaration just to be explicit here